lib/infer/params.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var types = require('ast-types');
+
+/**
+ * Infers param tags by reading function parameter names
+ *
+ * @name inferParams
+ * @param {Object} comment parsed comment
+ * @returns {Object} comment with parameters
+ */
+module.exports = function inferParams(comment) {
+
+  types.visit(comment.context.ast, {
+    visitFunction: function (path) {
+
+      // Ensure that explicitly specified parameters are not overridden
+      // by inferred parameters
+      var existingParams = comment.tags.filter(function (tag) {
+        return tag.title === 'param';
+      }).reduce(function (memo, param) {
+        memo[param.name] = param;
+        return memo;
+      }, {});
+
+      var paramOrder = [];
+
+      path.value.params.forEach(function (param) {
+        if (existingParams[param.name] === undefined) {
+          comment.tags.push({
```

Comment:
Sorry, I was unclear -- I meant you can drop the `comment.tags` stuff entirely in favor of `comment.params`, rather than in addition to it.